app/src/main/java/org/schabi/newpipe/Downloader.java

```diff
@@ -52,12 +65,41 @@
     private final OkHttpClient client;
 
     private Downloader(OkHttpClient.Builder builder) {
+        if(Build.VERSION.SDK_INT == Build.VERSION_CODES.KITKAT)
+            enableModernTLS(builder);
         this.client = builder
                 .readTimeout(30, TimeUnit.SECONDS)
                 //.cache(new Cache(new File(context.getExternalCacheDir(), "okhttp"), 16 * 1024 * 1024))
                 .build();
     }
 
+
+    // Enable TLS 1.2 and 1.1 for OkHTTP on Android Kitkat. This function is mostly taken from the documentation of OkHttpClient.Builder.sslSocketFactory(_,_)
```

Comment:
Please turn this to a valid Java doc comment by using `/**`.
And please stick to the java code styling conventions and add whitespaces before `{` and after `if` etc. This increases readability significantly and speeds up reviews :)

Reply:
Sure! is there a full styleguide available or profiles for android studio/linters? (I'm not that much into android dev)

Reply:
found it, i was just being stupid